components/layout_2020/flow/inline.rs

```diff
@@ -760,7 +759,12 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
             return;
         }
 
-        self.last_baseline_offset = Some(baseline_offset + block_start_position.into());
+        let baseline = baseline_offset + block_start_position;
+        if self.baselines.first.is_none() {
+            self.baselines.first = Some(baseline);
+        }
```

Comment:
Could be shorter with `self.baselines.first.get_or_insert(baseline);`, but either is fine.

Reply:
Sure. I've made this change.